src/ixdat/measurements.py

```diff
@@ -68,21 +69,22 @@ def __init__(
             reader (Reader): The file reader (None unless read from a file)
             plotter (Plotter): The visualization tool for the measurement
             exporter (Exporter): The exporting tool for the measurement
-            sample (Sample): The sample being measured
+            sample (Sample): The (already loaded) sample being measured
+            sample_name (str): The name of the sample being measured (will be loaded)
             lablog (LabLog): The log entry with e.g. notes taken during the measurement
             tstamp (float): The nominal starting time of the measurement, used for
                 data selection, visualization, and exporting.
         """
         super().__init__()
         self.name = name
         self.technique = technique
-        if isinstance(metadata, str):
```

Comment:
I think I actually kind of preferred the single argument multiple types form, simply because the ambiguity of what happens if you provide both disappear. But I must admit that I can come up with arguments both ways. What was your reason for changing it?

Another question for this is (I may have asked this before), will the json-string form ever be used by anything else than classmethods, because if not, they should convert to normal dict since it is unambiguous in their body and then the dual mode could disappear.

Reply:
It was an unresolved comment to #1 whether the responsibility of converting between dicts and json_strings (which at least the directory backend would need) belongs with the Saveable class or with the backend. 
https://github.com/ixdat/ixdat/pull/1#discussion_r546436991
I just saw your last comment there now (despite effort, I'm not getting the github notifications I want, must look into that again). Okay great, I am persuaded. `Measurement.metadata` will only exist as a dictionary. The directory backend will dump this to / load from a json string before saving/initiating. Other backends may do whatever.

Relevant, though, elsewhere in `ixdat`, is the one vs two arguments discussion:
The reason for two arguments is so that the type of any one of them is never ambiguous - I think a comment of yours in an earlier review encouraged me to try that (though maybe not right here).
Some clumsiness in these `.__init__`s in general I think has to do with the `Saveable` framework for class-table correspondence. The columns of tables now have to have the same names as the attributes that go into them.

Reply:
Done. Actually, `DirectoryBackend` simply dumps the `Saveable.as_dict` into json, and json has no problem with nested dictionaries, so `metadata` does not need to be converted to string at all for this backend.
Also, I left the `Measurement.metadata_json_string` property to remind anyone who sees it not to put unreadable stuff into `metadata`, but right now nothing uses the property.
It is not yet clear to me how another database type like mariaDB will know that it has to convert it to AND from json strings.
Also made `sample` one argument. But this is where it gets hairy, because what is stored in the db is `sample_name` whether `__init__()` is given `sample` as the Sample or as a string that it uses to load the sample. A similar problem is `DataSeries.unit` vs `DataSeries.unit_name`. One way (which I did first) to handle this is to make the Saveable class's `column_attrs` a dictionary rather than a set as it is now (see commit https://github.com/ixdat/ixdat/pull/5/commits/21017f40b9faadcb902fe60e387dece58148fbef for the conversion). But this is really clumsy and verbose, since most attributes are 1:1 with their column names.
I just implemented a new solution in `Measurement.from_dict()` which I agree is a reasonable place to handle this kind of thing. I've left a long comment there as well describing the problem and linking to this github comment.

Please resolve this and approve the PR if you think this is okay for now.

Reply:
I think this is a design issue that may warrant a little more discussion, but by all means let's move forward now.

As for both the MariaDB question and the `sample` problem I think that we need to go back to where `column_attrs` are a little more complex. So one way that I could imagine solving this is give a type for each column in `column_attrs` it could be trivial `int, bool, float` or something more complex like `json` but it could also be something custom, like `sample`. Each of the backends should then know, how to create the proper columns from all those types, including the custom ones, and how to convert back and forth.

Reply:
okay. Looking forward to more discussion!